### PR TITLE
Create Prime Number using C

### PR DESCRIPTION
**Pull request by manavbhikadiya**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by manavbhikadiya** [comment]
I have added prime number code in C.

File: C/math/primeNumber.c [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include <stdio.h>
+int main() {
+   int num = 33, flag = 0;
```
Comment (atarax665) [inline C/math/primeNumber.c:3]: Take input from the user.

File: C/math/primeNumber.c [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+   for(int i=2 ; i < num/2 ; i++) {
```
Comment (atarax665) [inline C/math/primeNumber.c:4]: The loop shouldn't iterate till ```n / 2 - 1```, optimize your code.

File: C/math/primeNumber.c [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      if(num%i == 0) {
+         printf("%d is not a prime number", num);
+         flag = 1;
+         break;
+      }
+   }
+   if(flag == 0) {
+      printf("%d is a prime number", num);
+   }
+}
```
Comment (atarax665) [inline C/math/primeNumber.c:14]: Add sample input/output and time complexity for your code in commented form. Refer any merged PR for better insight.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Add recursive approach for the same. And do the suggested changes.

**Pull request closed without merging**